packages/adapters-library/src/types/IProtocolAdapter.ts

```diff
@@ -20,6 +20,24 @@ import type {
 } from './adapter'
 import type { Erc20Metadata } from './erc20Metadata'
 
+export type Json =
+  | null
+  | boolean
+  | number
+  | string
+  | Json[]
+  | { [prop: string]: Json }
+
+export type JsonMetadata = Record<string, Json>
+
+export type ProtocolToken<
+  // biome-ignore lint/complexity/noBannedTypes: <explanation>
+  AdditionalMetadata extends JsonMetadata = {},
+> = Erc20Metadata & {
+  underlyingTokens?: Erc20Metadata[]
+  tokenId?: string
+} & AdditionalMetadata
```

Comment:
As well as having `underlyingTokens` and `tokenId` as optional, we could have two different types that extend from ProtocolToken that have them concretely. Something like `ProtocolTokenWithUnderlyings` and `ProtocolTokenNFT`.

Otherwise we have to keep specifying them as `AdditionalMetadata`, even though it's not really additional, just optional.

Reply:
On the other hand, we don't have to make that call now. What we store in itself won't change one way or another. So this is not a blocker.